db/migrate/20170515141252_add_custom_repositories.rb

```diff
@@ -0,0 +1,50 @@
+class AddCustomRepositories < ActiveRecord::Migration
+  def change
+    create_table :repositories do |t|
+      t.belongs_to :team, index: true
+      t.integer :created_by_id, null: false
+      t.string :name
+      t.timestamps null: true
+    end
+
+    add_foreign_key :repositories, :users, column: :created_by_id
+
+    create_table :repository_columns do |t|
+      t.belongs_to :repository, index: true
+      t.belongs_to :team, index: true
+      t.integer :created_by_id, null: false
+      t.string :name
+      t.integer :data_type, null: false
+      t.timestamps null: true
+    end
+
+    add_foreign_key :repository_columns, :users, column: :created_by_id
+
+    create_table :repository_rows do |t|
+      t.belongs_to :repository, index: true
+      t.belongs_to :team, index: true
+      t.integer :created_by_id, null: false
+      t.string :name
```

Comment:
It might make sense to index this column as well? We will allow people to sort over this column, right?